src/Nethermind/Nethermind.Db/SimpleFilePublicKeyDb.cs

```diff
@@ -122,15 +122,21 @@ private void CommitBatch()
             if (_logger.IsDebug) _logger.Debug($"Saving data in {DbPath} | backup stored in {backup.BackupPath}");
             try
             {
-                using StreamWriter streamWriter = new(DbPath);
+                using StreamWriter fileWriter = new(DbPath);
                 foreach ((byte[] key, byte[] value) in snapshot)
                 {
                     if (value is not null)
                     {
-                        key.StreamHex(streamWriter);
-                        streamWriter.Write(',');
-                        value.StreamHex(streamWriter);
-                        streamWriter.WriteLine();
+                        StringBuilder lineBuilder = new();
```

Comment:
very bad performance, to fix:
1. Reuse StringBuilder.
2. Write with:

`foreach (c in stringBuilder.GetChunks()) c.Span`

I expect this to have only one span almost always as it is small

3. Maybe initialize StringBuilder with some decently high capacity so it won't be chunked